spring-kafka/src/main/java/org/springframework/kafka/listener/ErrorHandlingUtils.java

```diff
@@ -256,4 +264,66 @@ public static Exception findRootCause(Exception exception) {
 		return realException;
 	}
 
+	/**
+	 * Determine whether the key or value deserializer is an instance of
+	 * {@link ErrorHandlingDeserializer}.
+	 * @param <K> the key type.
+	 * @param <V> the value type.
+	 * @param consumerFactory the consumer factory.
+	 * @param consumerOverrides consumer factory property overrides.
+	 * @param isValue true to find the value deserializer.
+	 * @param applicationContext the application context.
+	 * @return true if the deserializer is an instance of
+	 * {@link ErrorHandlingDeserializer}.
+	 * @since 3.0.10
+	 */
+	public static <K, V> boolean checkDeserializer(ConsumerFactory<K, V> consumerFactory,
+			Properties consumerOverrides, boolean isValue, @Nullable ApplicationContext applicationContext) {
```

Comment:
I don't have an application context in my hands, but I do configure a `ClassLoader`.
How about to change it to that one instead, please?